nilearn/decomposition/multi_pca.py

```diff
@@ -360,3 +372,44 @@ def inverse_transform(self, component_signals):
         # XXX: dealing properly with 2D/ list of 2D data?
         return [nifti_maps_masker.inverse_transform(signal)
                 for signal in component_signals]
+
+    def group_pca(self, subject_pcas):
+        """Compute the group PCA from individual subject PCAs"""
+        if self.incremental_group_pca:
+            return self._incremental_group_pca(subject_pcas)
+        else:
+            return self._in_memory_group_pca(subject_pcas)
+
+    def _incremental_group_pca(self, subject_pcas):
+        def _incremental_group_pca_impl(subject_pcas, n_components):
+            # delayed import because IncrementalPCA is only supported for
+            # sklearn versions > 0.16
+            from sklearn.decomposition.incremental_pca import IncrementalPCA
+
+            pca = IncrementalPCA(n_components=self.n_components)
+            for subject_pca in subject_pcas:
+                pca.partial_fit(subject_pca)
```

Comment:
> I think we should work out a way on lazy loading subject pcas : On HCP we
> typically use 100 components for 1650 records, which approximately reduce the
> whole data set of a factor 10 (1200 samples per records) : subject_pcas will be
> around 150GB, which is too much for a laptop.

The right thing to do is probably to use joblib shelving:
https://pythonhosted.org/joblib/memory.html#shelving-using-references-to-cached-values

Talk to @AlexandreAbraham: he has been using this in his related code.